src/main/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -2099,6 +2108,14 @@ public int hashCode() {
         return ret;
     }
 
+    private boolean isIndexingBinNotEqualsComputedBin() {
+        return this.getIndexingBin() != null && this.computeIndexingBin() != this.getIndexingBin();
+    }
+
+    private boolean isReferenceTooLargeForBAI(final SAMSequenceRecord sequence) {
+        return sequence.getSequenceLength() > GenomicIndexUtil.BIN_GENOMIC_SPAN;
+    }
+
```

Comment:
This method is duplicated here and in BAMRecordCodec. I think it would be useful to move it so it can be shared, maybe to GenomicUtils.

Reply:
Done.